cassandane/Cassandane/Cyrus/JMAPMailbox.pm

```diff
@@ -5282,4 +5296,64 @@ sub test_mailbox_ignore_notes_subfolders
     $self->assert_str_equals($inboxId, $res->[1][1]{list}[0]{id});
 }
 
+sub test_mailbox_set_create_specialuse_nochildren
+    :min_version_3_7 :needs_component_jmap :NoStartInstances
+{
+    my ($self) = @_;
+
+    $self->{instance}->{config}->set('specialuse_nochildren' => '\\Trash');
```

Comment:
Why not introduce a new test magic for this? Or even put it into the default config? It looks as if allowing sub mailboxes in Trash is the non-standard case, and this would allow us to catch any other tests that break this assumption.

Reply:
The default value for specialuse_nochildren is `\Scheduled \Snoozed`, which doesn't include \Trash.  If we set \Trash in the default config, then every test that only expects the default set would then need some override to remove \Trash.

I could add a magic for it, but since it's only used by two tests, it doesn't feel worth the effort yet -- my usual threshold is when I find myself doing the same thing the same way a third time, which I haven't yet.  It's tricky anyway, because -- should the magic set "only \Trash" (as these tests do, which don't care about \Scheduled or \Snoozed), or "add  \Trash to the set"?  And will we remember to update the magic in the future if the default set changes?  Until we need it a third time, and know what shape of thing we regularly need, I don't think it's worth the thought.

When we run the tests from in-a-box with the Fastmail config in cassandane.ini, it should have \Trash included in the set because that's the configuration we use, so that will catch any tests that are making bad assumptions.

Reply:
My threshold for adding new test magics is considerably lower, but I do not object to the current implementation.